src/box/arrow_ipc.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2024, Tarantool AUTHORS, please see AUTHORS file.
+ */
+#pragma once
+
+#include "arrow/abi.h"
+
+#if defined(__cplusplus)
+extern "C" {
+#endif /* defined(__cplusplus) */
+
+struct arrow_ipc_buffer;
+
+/** Allocates and initializes an Arrow IPC buffer, can not fail. */
+struct arrow_ipc_buffer *
+arrow_ipc_buffer_new(void);
+
+/** Destroys and deallocates an Arrow IPC buffer. */
+void
+arrow_ipc_buffer_delete(struct arrow_ipc_buffer *ipc_buffer);
+
+/**
+ * Encodes `array' and `schema' into Arrow IPC format. The memory is allocated
+ * by the `ipc_buffer', it must be initialized by `arrow_ipc_buffer_new()'.
```

Comment:
I don't understand the purpose of this buffer in the API. Maybe, better make it static?

Reply:
Removed external buffer, changed to:

```c
        /*
         * TODO: It is possible to avoid extra `memcpy()' by switching
         * `ArrowBuffer' to `region_realloc()'.
         */
        char *data = xregion_alloc(region, buffer.size_bytes);
        memcpy(data, buffer.data, buffer.size_bytes);
```